src/sentry_sync.c

```diff
@@ -21,10 +33,11 @@ struct sentry_bgworker_s {
     sentry_mutex_t done_signal_lock;
     sentry_mutex_t task_lock;
     sentry_threadid_t thread_id;
-    struct sentry_bgworker_task_s *first_task;
-    struct sentry_bgworker_task_s *last_task;
+    sentry_bgworker_task_t *first_task;
+    sentry_bgworker_task_t *last_task;
     size_t task_count;
     bool running;
+    bool free_self;
```

Comment:
The entire bgworker state needs to be owned by both the main thread (where `sentry_init` and `sentry_shutdown` is run), and the background worker. Thus, we need to employ refcounting.

Once we release both, it can clean up, which replaces `sentry__bgworker_free` with `sentry__bgworker__decref`. The nice side effect of this is that `decref` no longer needs to lock :)